addons/microsoft_calendar/models/microsoft_sync.py

```diff
@@ -417,7 +417,8 @@ def _microsoft_patch(self, user_id, event_id, values, timeout=TIMEOUT):
         due to @after_commit decorator.
         """
         microsoft_service = self._get_microsoft_service()
-        with microsoft_calendar_token(self._impersonate_user(user_id).sudo()) as token:
+        sender_user = self.env.user if not user_id._get_microsoft_calendar_token() else user_id
```

Comment:
Prefer using direct logic rather than negating stuff
```suggestion
sender_user = user_id if user_id._get_microsoft_calendar_token() else self.env.user
```